runner/config.ts

```diff
@@ -128,6 +145,97 @@ export function getPackageName(options: Config): string|undefined {
   return basename;
 }
 
+/**
+ * Finds the first npm scope starting with `@` from a path. Undefined if no path
+ * found
+ *
+ * @param path Path to be searched
+ */
+export function getPackageScopeFromPath(path: string): string|undefined {
+  const pathParts = path.split('/');
+
+  for (const pathPart of pathParts) {
+    if (pathPart[0] === '@') {
+      return pathPart;
+    }
+  }
+
+  return undefined;
+}
+
+/**
+ * Truncates the path to the slash after the last occurrence of the given
+ * package name.
+ *
+ * @param directory Name of directory
+ * @param path Path to be truncated
+ */
+export function truncatePathToDir(directory: string, path: string): string|
+    null {
+  const delimitedDir = `/${directory}/`;
+  const lastDirOccurrence = path.lastIndexOf(delimitedDir);
+  if (lastDirOccurrence === -1) {
+    return null;
+  }
+  return path.substr(0, lastDirOccurrence + delimitedDir.length);
+}
+
+/**
+ * Resolves npm paths from current config root to ScriptNames.
+ *
+ * e.g. a/b.js is actually resolved in directory c's node modules, it would
+ * return c/node_modules/a/b.js. These dependencies must be direct dependencies
+ * to WCT.
+ *
+ * @param config Current config / options / scope
+ * @param npmPackages List of NPMScript objects to be resolved
+ * @param wctPackageName Name of wct package with browser.js (Defaults to
+ * wct-browser-legacy)
+ */
+export function resolveWCTNPMEntrypointNames(
+    config: Config, npmPackages: NPMPackage[],
+    wctPackageName = 'wct-browser-legacy'): string[] {
+  // grab from CLI flag defaults to wct-browser-legacy
+  wctPackageName =
+      config.wctPackageName ? config.wctPackageName : wctPackageName;
+  let absoluteBrowserPath;
+
+  try {
+    console.log(`${npmPackages[0].name}
+    basedir: ${config.root}
+    package name: ${wctPackageName}`);
+    absoluteBrowserPath = resolve.sync(wctPackageName, {basedir: config.root});
+  } catch {
+    throw new Error(
+        `${wctPackageName} not installed. Please change --wct-package-name` +
+        ` flag or install the package.`);
+  }
+
+  // We want to find and inject dependencies WCT relies on not the local
+  // package or its dependencies' dependencies
+  const absoluteWCTRoot =
```

Comment:
the async version of `resolve()` has a `pathFilter` option that might let you get the package dir without the `truncatePathToDir` function: https://github.com/browserify/resolve#resolveid-opts-cb

I think you can return the empty string and get what you want, and you can use the async version because `resolveWCTNPMEntrypointNames` is called from an async function.

Reply:
I can't seem to get `pathFilter` to even get called (a me problem probs). Regardless, it seems as if `resolve()` returns undefined rather than a promise.

Reply:
Trying this again and I simply cannot get `pathFilter` to work. Running this in node from the `test/fixtures/fake-packages/duplicated-dep` directory

```js
resolve('dependency', {
  pathFilter: function(pkg, path, relPath) {
    console.log('PFILTER ARGS:',arguments);
    return path;
  }
},
function() { console.log(`CB ARGS:`, arguments) });
```

it just wont call, and we're using the latest release of `resolve`